src/azure-cli/azure/cli/command_modules/role/_help.py

```diff
@@ -361,6 +361,7 @@
 helps['ad sp create'] = """
 type: command
 short-summary: Create a service principal.
+long-summary: When you create a service principal using the `az ad sp create` command, the output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. As an alternative, consider using `managed identities` (https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) to avoid the need to use credentials.
```

Comment:
`ad sp create` simple creates a Service Principal from an app and won't show any credentials.

```
> az ad app create --display-name myapp20201110
  ...
  "appId": "d990dd3e-f5fe-4d3c-bb62-3e568a5b1209",

> az ad sp create --id d990dd3e-f5fe-4d3c-bb62-3e568a5b1209
  ...
  "appId": "d990dd3e-f5fe-4d3c-bb62-3e568a5b1209",
```

This help message should go to `az ad app/sp credential reset` instead. 
```
> az ad app credential reset --id d990dd3e-f5fe-4d3c-bb62-3e568a5b1209
{
  "appId": "d990dd3e-f5fe-4d3c-bb62-3e568a5b1209",
  "name": "d990dd3e-f5fe-4d3c-bb62-3e568a5b1209",
  "password": "3f~XjdIB0AML4...",
  "tenant": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a"
}
```